test/python/WMCore_t/FwkJobReport_t/Report_t.py

```diff
@@ -584,6 +588,27 @@ def testExitCode(self):
         self.assertItemsEqual(report.getStepExitCodes(stepName="cmsRun1"), {99999, 12345, 123})
         self.assertEqual(report.getStepErrors(stepName="cmsRun1")['errorCount'], 3)
 
+        # and once again, but different type and details (which does not matter) - testing unicode handling
+        report.addError(stepName="cmsRun1", exitCode=12345, errorType="⊥ℯṧтÅÅ", errorDetails="тℯṧт")
+        self.assertEqual(report.getExitCode(), 12345)
+        self.assertEqual(report.getStepExitCode(stepName="cmsRun1"), 12345)
+        self.assertItemsEqual(report.getStepExitCodes(stepName="cmsRun1"), {99999, 12345, 123})
+        self.assertEqual(report.getStepErrors(stepName="cmsRun1")['errorCount'], 3)
+
+        # and once again, but different type and details (which does not matter) - testing unicode handling
+        report.addError(stepName="cmsRun1", exitCode=12345, errorType=decodeBytesToUnicode("⊥ℯṧтÅÅ"), errorDetails=decodeBytesToUnicode("тℯṧт"))
+        self.assertEqual(report.getExitCode(), 12345)
+        self.assertEqual(report.getStepExitCode(stepName="cmsRun1"), 12345)
+        self.assertItemsEqual(report.getStepExitCodes(stepName="cmsRun1"), {99999, 12345, 123})
+        self.assertEqual(report.getStepErrors(stepName="cmsRun1")['errorCount'], 3)
+
+        # and once again, but different type and details (which does not matter) - testing unicode handling
+        report.addError(stepName="cmsRun1", exitCode=12345, errorType=encodeUnicodeToBytes("⊥ℯṧтÅÅ"), errorDetails=encodeUnicodeToBytes("тℯṧт"))
+        self.assertEqual(report.getExitCode(), 12345)
+        self.assertEqual(report.getStepExitCode(stepName="cmsRun1"), 12345)
+        self.assertItemsEqual(report.getStepExitCodes(stepName="cmsRun1"), {99999, 12345, 123})
+        self.assertEqual(report.getStepErrors(stepName="cmsRun1")['errorCount'], 3)
+
```

Comment:
Trying to pass both unicode and bytes strings containing non-ascii chracters into `Report.addError`, both in py2 and py3

Reply:
Improved in later commit, marking this as resolved